tests/unit/connection.js

```diff
@@ -0,0 +1,228 @@
+const config = require('../data/config')
+const Connection = require('../../lib/connection')
+const helper = require('../helper')
+const sinon = require('sinon')
+
+exports['static - withConnection'] = {
+  setUp (done) {
+    helper.initDatabase.call(this).then(() => done())
+  },
+
+  tearDown (done) {
+    this.connection.end()
+    done()
+  },
+
+  'error with connect' (test) {
+    const providedError = new Error()
+    sinon.stub(Connection.prototype, 'connect').returns(Promise.reject(providedError))
+
+    Connection.withConnection(config, (connection) => {
+      test.ok(false, 'should not invoke callback')
+    })
+      .catch((error) => {
+        test.strictEqual(error, providedError, 'should propagate error')
+
+        Connection.prototype.connect.restore()
+        test.done()
```

Comment:
Sure, but the next PR will make the tests work on Travis. See https://github.com/clipper-digital/promiscuous-mysql/pull/1/files/34bf8679703ed69bbf0e7d1b079a48ec67ce6956#r78419431